File: mqtt-manager/src/main/java/com/ns/greg/library/mqtt_manager/external/Qos.java
package com.ns.greg.library.mqtt_manager.external;

import com.ns.greg.library.mqtt_manager.MqttManager;

/**
 * @author Gregory
 * @since 2017/12/22
 */

public interface Qos {

  @MqttManager.QoS int getQoS();
}

File: mqtt-manager/src/main/kotlin/com/ns/greg/library/mqtt_manager/external/QoS.kt
package com.ns.greg.library.mqtt_manager.external

/**
 * @author gregho
 * @since 2019-06-20
 */
enum class QoS(val value: Int) {

  AT_MOST_ONCE(0),
  AT_LEAST_ONCE(1),
  EXACTLY_ONCE(2)
}